tests/conftest.py

```diff
@@ -59,15 +59,19 @@ async def client(app: FastAPI) -> AsyncIterator[AsyncClient]:
 
 
 @pytest.fixture
-def jupyterhub(mock_aioresponses: aioresponses) -> Iterator[None]:
-    """Mock out JupyterHub."""
-    mock_jupyterhub(mock_aioresponses)
+def jupyter(mock_aioresponses: aioresponses) -> Iterator[None]:
+    """Mock out JupyterHub/Lab."""
+    mock_jupyter(mock_aioresponses)
 
     # aioresponses has no mechanism to mock ws_connect, so we can't properly
     # test JupyterClient.run_python.  For now, just mock it out entirely.
     with patch.object(JupyterClient, "run_python") as mock:
         mock.return_value = "4"
-        yield
+        # Same problem, but up a layer now that we're using sessions and
+        # reusing the websocket.
+        with patch.object(JupyterClient, "_websocket_connect") as mock2:
+            mock2.return_value = None
```

Comment:
If you return `MagicMock()` instead, you can get rid of the `Optional` on this value, which in turn simplifies some other code and removes the need for the `None` check.

```suggestion
            mock2.return_value = MagicMock()
```

and import `MagicMock` from `unittest.mock`.

Reply:
Alas, no, then I get 
    await session.websocket.close()
TypeError: object MagicMock can't be used in 'await' expression
ERROR    testuser1:monkey.py:79 [error    ] Slack Alert: 2021-07-27 13:32:06 testuser1 object MagicMock can't be used in 'await' expression [testuser1]
﻿

Reply:
But there's AsyncMock!

Reply:
AsyncMock works.  Done!